src/ast/query.rs

```diff
@@ -1865,27 +1873,57 @@ impl fmt::Display for SelectInto {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub enum WithModifier {
```

Comment:
```suggestion
pub enum GroupByWithModifier {
```
thinking we can mention group by in the name to make it unambiguous?